api/lib/opentelemetry/distributed_context/propagation/http_text_format.rb

```diff
@@ -17,7 +16,35 @@ module Propagation
       # Propagation is usually implemented via library-specific request interceptors, where the client-side injects values
       # and the server-side extracts them.
       class HTTPTextFormat
-        # TODO
+        # extract will return a SpanContext from the supplied carrier
+        # invalid headers will result in a new SpanContext
+        # @param [Carrier] the carrier to get the header from
+        # @yield [Carrier, String] the header key
+        # {SpanContext}
+        def extract(carrier)
+          raise Error, 'block must be supplied' unless block_given?
```

Comment:
I think this should be an ordinary `ArgumentError` rather than the OT `Error`. Same thing below in `inject`.